Dockerfile.perses-operator

```diff
@@ -0,0 +1,15 @@
+FROM alpine AS build-env
+RUN apk add --update --no-cache mailcap
+
+FROM gcr.io/distroless/static-debian12
+
+LABEL maintainer="The Perses Authors <perses-team@googlegroups.com>"
+
+USER nobody
+
+COPY --chown=nobody:nobody perses-operator/bin/manager        /bin/manager
```

Comment:
This image asumes the binary is already built, we need to change it to one that actually builds the binary

Reply:
Update this file to mimic [/perses/perses-operator/Dockerfile.dev](https://github.com/perses/perses-operator/blob/main/Dockerfile.dev)

Updated to use equivalent redhat validated images. I think Dockerfile.obo is using similar images so I used those: 

Before 
`FROM golang:1.23-alpine AS build-env`
After 
`FROM brew.registry.redhat.io/rh-osbs/openshift-golang-builder:rhel_8_golang_1.22 as build-env`

Before 
`FROM gcr.io/distroless/static-debian12`
After 
`FROM registry.redhat.io/ubi8/ubi-minimal:8.10-1130`

Reply:
Update this file to mimic [/perses/perses-operator/Dockerfile.dev](https://github.com/perses/perses-operator/blob/main/Dockerfile.dev)

Updated to use equivalent redhat validated images. I think Dockerfile.obo is using similar images so I used those: 

Before 
`FROM golang:1.23-alpine AS build-env`
After 
`FROM brew.registry.redhat.io/rh-osbs/openshift-golang-builder:rhel_8_golang_1.22 as build-env`

Before 
`FROM gcr.io/distroless/static-debian12`
After 
`FROM registry.redhat.io/ubi8/ubi-minimal:8.10-1130`